src/main/java/org/springframework/data/repository/config/RepositoryBeanDefinitionBuilder.java

```diff
@@ -128,8 +131,9 @@ public BeanDefinitionBuilder build(RepositoryConfiguration<?> configuration) {
 
 		fragmentsBuilder.addConstructorArgValue(fragmentBeanNames);
 
-		builder.addPropertyValue("repositoryFragments",
-				ParsingUtils.getSourceBeanDefinition(fragmentsBuilder, configuration.getSource()));
+		String fragmentsBeanName = BeanDefinitionReaderUtils.uniqueBeanName(extension.getModuleName().toLowerCase(Locale.ROOT) + ".repo-fragments", registry);
```

Comment:
It would make sense to include the repository name for easier correlation as the fragments are correlated to an individual repository.